src/config.rs

```diff
@@ -97,46 +200,60 @@ impl ConfigStore {
     }
 
     /// Returns the setting with the given key. If the setting is not found in the current
-    /// store, it will load the key from the storage. If the key is still not found, it will
+    /// storage, it will load the key from the storage. If the key is still not found, it will
     /// return the default value for the given key. Note that if the key is not found and no
     /// default value is specified, this function will panic.
-    pub fn get(&mut self, key: &str) -> Setting {
+    pub fn get(&self, key: &str) -> Option<Setting> {
         if !self.has(key) {
-            panic!("Setting {} not found", key);
+            return None;
```

Comment:
This was a panic, but we just return None.. it's up to the caller to deal with it

Reply:
This might indeed be false. I'll check

Reply:
fixed this. This flow was not correct after some refactoring here and there.. surprised it still passed the tests